### ISSUE TITLE
Blokada adresu IP - 504

### ISSUE THREAD

**Issue by MarcinTroska** [allegro/allegro-api#9672]
Występuje problem:
 - [X ] z zasobem REST API / with a REST API resource
 - [ X] ze środowiskiem produkcyjnym / with the production environment
 - [ ] ze środowiskiem testowym / with the test environment
 - [X] z logowaniem lub autoryzacją / with login or authorization
 - [ ] na portalu https://developer.allegro.pl/ / at https://developer.allegro.pl/en/
 - [ ] inny / other

:beetle: **Opis / Description**
Witam, dnia 19.08.2024 około godziny 8.00 (czasu PL)  otrzymaliśmy na wszystkich możliwych endpointach response 504. Chciałem poznać powód blokady i go wyeleminować po naszej stronie

```
<html>                                                                                                                                                                                          
       <head><title>504 Gateway Time-out</title></head>                                                                                                                                                             
        <body> <center><h1>504 Gateway Time-out</h1></center>  <hr><center>nginx</center> </body>                                                                                                                                                                                                       </html>  
```                                                                                                                                                                                               

zblokowany adres IP 20.101.166.13

**Comment by MaciejFrackowiak** [comment]
Proszę podaj trace-id przykładowego wywołania, ponieważ nie widzę błędnych wywołań z podanego adresu IP.

**Comment by MarcinTroska** [comment]
[logi.txt](https://github.com/user-attachments/files/16657795/logi.txt) - to jest jedyne co mam na ten moment. EP które, przestały odpowiadać to min 

1. https://api.allegro.pl/order/event-stats
2. https://api.allegro.pl/sale/shipping-rates
3. https://api.allegro.pl/sale/product-offers
4. ttps://api.allegro.pl/order/checkout-forms/

**Comment by MaciejFrackowiak** [comment]
Proszę podaj mi ClientID aplikacji, która wywołuje zasoby.

**Comment by MarcinTroska** [comment]
fe1560d9f7e84eea923ab00e208d4500 - podsyłam clientId

**Comment by MaciejFrackowiak** [comment]
Nie widzę żadnych błędów 50x dla wskazanej aplikacji w ostatnich godzinach. Co więcej, od 9:16 widzę kolejne bieżące wywołania zakończone powodzeniem - kod 200.

**Comment by MaciejFrackowiak** [comment]
Jak wspomniałem nie mamy śladu po jakichkolwiek błędnych zapytaniach wskazanej aplikacji. Bez konkretnych danych tj. trace-id wywołań praktycznie nic więcej ustalić nie możemy.

**Comment by MarcinTroska** [comment]
Dziękuje w takim razie za pomoc. Przygotuje sobie odpowiednie dane, by w razie powrotu błędu będę mógł wrócić z konkretami

**Comment by MarcinTroska** [comment]
Dzień dobry, widzę, że issue nie jest zamknięte. Problem powrócił dnia 11.10.2024 w godzinach +/- 16.40. Tak, jak poprzednio - przez pewien czas nie API Allegro zwracało 504.  

Poniżej lista trace-id, Mam nadzieję, że teraz uda nam się namierzyć problem:
  
00-9b537daae63b9a983750667335c5982e-c6af91c04931e2a0-00
00-64e7fcf245ceda591b39320ca5aa07d7-a47f9318774d11e3-00
00-7dade4d949d891d215da48163369dfb5-73dd2d14c1d4647b-00
00-fb1e0bf3fbf06939bef3128b35cc128a-5b2ba96c7bf88747-00
00-bc0cae43750a7ab0c9ee53fa00428bff-1067fae777e3b019-00
00-d9929bd2f61c24655074fc914c3511fc-27bc4f78db18c34f-00
00-1cb84d773360e0cb888a051ba026bce1-f6ed26f461722dd8-00
00-d1b771fd50d5f8c413aa418dd96029ce-078750d0998774da-00
00-55a71f035a5c5c8dfead2f84e8524ac6-20ea7801330b2e2b-00
00-7bcc02055822e598774154aabce08f56-c87c6a0795162afa-00
00-e488834bb946c070ce897875e61a93b1-d15b56368f1cd6a1-00
00-62d63f836d98aec9fab9b3005d9746a1-52665a46cbece687-00
00-04c7ad11dfd2f1e1e22eb2ebba6654f2-f8e8e9cf39c3a548-00
00-e0bb30787a6671781d057fab245cec06-507ee535e91f0c09-00
00-309800670a2ac1b497e025821ddd6c82-a57ee28bfb6d08b3-00
00-336c6effef8d32a05ca5f9cfc74199f3-69589f3985f23762-00
00-c01a052b10505c0d952ebc327cae3f21-abc0892fcc220680-00
00-742a031ee38a95ebf6aff35e19e6b92b-69ef433513ab707c-00
00-3e54f78eeca667fce35dadabc776d780-e0e67fc8af15d922-00
00-c2b8521d26b73206a2c6e1c287a6e996-2435e4689f7d3145-00
00-b3afd3d138bcab21eef7c22d45ccb437-af925c14d0c7ec5c-00
00-f1275d19bf2f57c88e8a8a358b85323b-59c55389a679be51-00
00-8b32b5f0efebfd26d165152c9e5cac7c-3e69ec232e4f33bc-00

**Comment by MaciejFrackowiak** [comment]
Niestety, ale to nie są trace-id zwracane przez nas bezpośrednio. Bez właściwych przykładów nie mogę niczego sprawdzić.

**Comment by MarcinTroska** [comment]
Panie Macieju, posiadamy tutaj GateWay Time-Out i zakładam, że nie dochodzimy do miejsca, w którym Pan rozpoznje trace-id np. poprawny jaki otrzymujemy "ee4efcaef3927f14". Czy te 504 nie wynikają z jakiś Load Balancerów lub innego systemu po drodze do api.allegro.pl ?

**Comment by MaciejFrackowiak** [comment]
Nie, przedstawiony format trace-id nie wskazuje na żadne z naszych narzędzi.
